### PR TITLE
#181339541-trip-comment

### PR DESCRIPTION
**Pull request by Code-250**
### what does this Pull Request do?
this Pull Request implements commenting on a trip and deleting a comment

### description of tasks to be completed
 - comment on a trip
 - delete a comment of a trip only performed by the owner of the comment
 - listing/ viewing all comments on a specific trip

### how should this Pull Request be tested?

 - checkout `feature/#181339541-trip-comments` branch
 - run in your terminal `docker-compose up` to start the docker containers and the project.
 - then run `docker exec -it cabal-backend sh` to inside the container
 - run `npx sequelize db:migrate ` and `npxsequelize db:seed:all` to migrate new tables in your database.

### pivotal tracker
 - [trip-comments](https://www.pivotaltracker.com/n/projects/2555280/stories/181339541)

### test in production
[swagger-Doc](https://staging-bare-feature-18-l5lzge.herokuapp.com/api-docs/)
N/A

### CODE REVIEW

File: src/controllers/trip.comments.controller.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  static async findAllComments(req, res) {
+    try {
+      const found = await tripCommentsServices.findTripComments();
+      if (!found)
+        return res.status(404).json({ message: 'comments does not exist' });
+      return res.status(200).json({ message: 'retrieved all comments', found });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
```
Comment (Tsa2341) [inline src/controllers/trip.comments.controller.js]: Let the developer know the error in all catch if unexpected

File: src/controllers/trip.comments.controller.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      );
+      const findEdited = await tripCommentsServices.findSpecificComment(
+        commentId
+      );
+
+      return res
+        .status(200)
+        .json({ message: 'successfully edited a comment', findEdited });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
```
Comment (Tsa2341) [inline src/controllers/trip.comments.controller.js]: Let the developer know the error in all catch if unexpected

File: src/controllers/trip.comments.controller.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          .json({ message: 'you are not allowed to delete this comment' });
+      const destroyComment = await tripCommentsServices.removeComment({
+        where: { id: commentId }
+      });
+      return res
+        .status(200)
+        .json({ message: 'successfuly removed a comment', destroyComment });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error' });
+    }
```
Comment (Tsa2341) [inline src/controllers/trip.comments.controller.js]: Let the developer know the error in all catch if unexpected

please do so in all the other catches.

**Review by Tsa2341 — CHANGES_REQUESTED** [review]
Also add the swagger or deployment link in your pull request

**Review by Tsa2341 — APPROVED** [review]

**Review by blandri — APPROVED** [review]

**Review by niyonkurueric — APPROVED** [review]

**Review by Gabin-ishimwe — APPROVED** [review]

**Comment by ncutixavier** [comment]
### Comments

- Your PR has not been deployed
- Try to raise your test coverage to be above 85%

**Comment by ncutixavier** [comment]
Squash your commit

**Comment by Code-250** [comment]
commits are squashed now

File: src/controllers/trip.comments.controller.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      const { user } = req;
+      const foundComment = req.comment.dataValues;
+      const commentId = req.comment.dataValues.id;
+      if (
+        user.dataValues.id !== req.trip.dataValues.user_id &&
+        user.dataValues.manager_id !== req.trip.dataValues.manager_id
+      )
+        return res.status(409).json({
+          message: 'you are not allowed to perform this activity'
+        });
```
Comment (ncutixavier) [inline src/controllers/trip.comments.controller.js:91]: Change status code

File: src/controllers/trip.comments.controller.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        /* istanbul ignore next */
+      if (user.dataValues.id !== foundComment.user_id)
+        return res
+          .status(409)
+          .json({ message: 'you are not allowed to delete this comment' });
```
Comment (ncutixavier) [inline src/controllers/trip.comments.controller.js]: Change status code

File: tests/accommodation.test.js [hunk 6]
```diff
@@ -313,6 +467,56 @@ describe('ACCOMMODATION ENDPOINT TESTING', () => {
...
+        description:
+          'In publishing and graphic design, Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a typeface without relying .',
+        location_id: 13,
+        services: ['restaurant', 'breakfast', 'gym', 'swimming pool'],
+        amenities: ['restaurant', 'breakfast', 'gym', 'swimming pool'],
+        images: [
+          'https://images.unsplash.com/photo-1522798514-97ceb8c4f1c8?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTF8fGhvdGVsfGVufDB8fDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60'
+        ],
+        imagesId: ['456780rty']
+      });
```
Comment (ncutixavier) [inline tests/accommodation.test.js]: Refactor this code

File: tests/accommodation.test.js [hunk 7]
```diff
@@ -358,4 +562,71 @@ describe('ACCOMMODATION ENDPOINT TESTING', () => {
...
+      .send({
+        name: 'Serena Hotel',
+        description:
+          'In publishing and graphic design, Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a typeface without relying .',
+        location_id: 15,
+        services: ['restaurant', 'breakfast', 'gym', 'swimming pool'],
+        amenities: ['restaurant', 'breakfast', 'gym', 'swimming pool'],
+        image:
+          'https://images.unsplash.com/photo-1522798514-97ceb8c4f1c8?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTF8fGhvdGVsfGVufDB8fDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60',
+        imagesId: ['456780rty']
```
Comment (ncutixavier) [inline tests/accommodation.test.js]: Refactor this code

File: tests/user.test.js [hunk 8]
```diff
@@ -154,20 +155,20 @@ describe('USER END-POINT TEST', () => {
...
-        });
-      const { token } = res.body;
+    // it('should validate user email', async () => {
+    //   const res = await request(app)
+    //     .post('/api/v1/users/register')
+    //     .send({
+    //       email: `T${new Date().getMilliseconds()}test@gmail.com`,
+    //       password: 'Tester12345'
+    //     });
+    //   const { token } = res.body;
```
Comment (ncutixavier) [inline tests/user.test.js]: Remove commented code

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Pull request merged**